components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/webpush/GeckoWebPushDelegate.kt

```diff
@@ -22,11 +22,7 @@ internal class GeckoWebPushDelegate(private val delegate: WebPushDelegate) : Gec
         val result: GeckoResult<GeckoWebPushSubscription> = GeckoResult()
 
         delegate.onGetSubscription(scope) { subscription ->
-            if (subscription != null) {
-                result.complete(subscription.toGeckoSubscription())
-            } else {
-                result.completeExceptionally(WebPushException("Retrieving subscription failed."))
-            }
+            result.complete(subscription?.toGeckoSubscription())
```

Comment:
Due to `connection.ifInitialized` this may never get called, I think. That seems like a problem, if we never complete the result.

Reply:
Answered below.